devito/dse/rewriters.py

```diff
@@ -406,6 +408,38 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        A pass to perform only the basic skewing in order to have valid data
+        dependences.
+        """
+        skew_factor = configuration['skew_factor']
+        skew_dim, mapper, int_mapper = None, {}, []
+        total_int = {}
+        intervals, sub_iterators, directions = cluster.ispace.args
+
+        for dim in cluster.ispace.dimensions:
```

Comment:
I think you want to iterate over ``cluster.ispace.intervals`` (or even ``cluster.itintervals``)

then you can get the ``dim`` via ``i.dim``